packages/cloudapi-gql/src/schema/resolvers.js

```diff
@@ -58,15 +58,24 @@ const resolvers = {
     package: (root, { id, name }) => api.packages.get({ id, name }),
 
     machines: (root, { id, brand, state, tags, ...rest }) =>
-      id
+      id 
         ? api.machines.get({ id }).then(machine => [machine])
         : api.machines.list(
             Object.assign(rest, {
               brand: brand ? brand.toLowerCase() : brand,
               state: state ? state.toLowerCase() : state,
-              tags: fromKeyValue(tags)
+              tags: fromKeyValue(tags),
             })
-          ),
+          ).then(machines =>
+            machines.map(machine => {
```

Comment:
can we only fetch each machine again only and only if the dns_names prop is requested?

Reply:
Sure , how do I do that ? :(

Reply:
https://media.giphy.com/media/Ob7p7lDT99cd2/giphy.gif